.pre-commit-config.yaml

```diff
@@ -67,8 +70,39 @@ repos:
       - id: flake8
         additional_dependencies:
           - flake8-bugbear
+-   repo: https://github.com/pre-commit/mirrors-mypy
+    rev: 'v0.991'
+    hooks:
+    -   id: mypy
+        files: |
+            (?x)^src/snowflake/connector/(
+                constants
+                | compat
+                | dbapi
+                | description
+                | errorcode
+                | feature
+                | file_compression_type
+                | file_util
+                | gzip_decoder
+                | proxy
+                | secret_detector
+                | sfbinaryformat
+                | snow_logging
+                | sqlstate
+                | ssd_internal_keys
+                | test_util
+                | util_text
+                | version
```

Comment:
what's the list here.. is that we only apply mypy check to the listed file? I didn't see cache/ocsp_snowflake etc. being listed here.